facebook/react-native
# [Android] Remove Stetho dependency from ReactAndroid

**Fixes**: #6131 
Stetho is a debug dependency which should not be included in releases. It unnecessarily bloats the apk size.

This change removes the dependency from the ReactAndroid library and adds an example Stetho implementation to the Android template. 

### Test plan
1. `react-native init StethoExample`
2. Update `package.json` to use `git@github.com:facebook/react-native.git#master`
3. `npm install` & `react-native upgrade`
4. Clone `https://github.com/AndrewJack/react-native/tree/remove_stetho_from_react_android`
5. Update the `android/build.gradle` file as specified in the building from source [guide](https://facebook.github.io/react-native/docs/android-building-from-source.html). Specify the ReactAndroid folder in the repo you just cloned.
6. Update `android/app/build.gradle` dependencies to use the module ReactAndroid. `compile project(':ReactAndroid')`
7. Configure Stetho to only compile with debug by using `debugCompile`
2. `cd StethoExample/android && ./gradlew app:assembleRelease`
3. Find APK in `StethoExample/android/app/build/outputs/apk`
4. Open apk in [ClassyShark](https://github.com/google/android-classyshark)
5. See Stetho not included in APK
